ens_mean_recenter/module_mpi_arrange.f90

```diff
@@ -142,174 +142,139 @@ subroutine arrange(this,ncfs_dyn,ncfs_tracer,ncfs_sfc)
-           endif
-        enddo
-    endif
-
-    if(ncfs_sfc%numvar > 0) then
-        do i=1,ncfs_sfc%numvar
-           if(ncfs_sfc%num_dim(i)==3) then
-              nz=ncfs_sfc%dim_3(i)
-              n3d=n3d+1
-              mynlvl3d=min(nlvl3d_list(n3d),nz)
-              nlvlcore=nz/mynlvl3d
-              j=nz-nlvlcore*mynlvl3d
-              do k=1,mynlvl3d
-                 nn=nn+1
-                 fileid(nn)=3
-                 varname(nn)=trim(ncfs_sfc%list_varname(i))
-                 vartype(nn)=ncfs_sfc%vartype(i)
-                 nx(nn)=ncfs_sfc%dim_1(i)
-                 ny(nn)=ncfs_sfc%dim_2(i)
-                 if(k==1) then
-                    lvlbegin(nn)=1
-                 else
-                    lvlbegin(nn)=lvlend(nn-1)+1
-                 endif
-                 lvlend(nn)=lvlbegin(nn)+nlvlcore-1
-                 if(k<=j) lvlend(nn)=lvlend(nn)+1
-                 if(k==mynlvl3d) lvlend(nn)=min(lvlend(nn),nz)
-              enddo
-           elseif(ncfs_sfc%num_dim(i)==2) then
+           if(ncfs_all%num_dim(i)==3) then
+              nz=ncfs_all%dim_3(i)
+              if(nz <= 10) then
+                 do k=1,nz
+                    nn=nn+1
+                    varname(nn)=trim(ncfs_all%list_varname(i))
+                    vartype(nn)=ncfs_all%vartype(i)
+                    nx(nn)=ncfs_all%dim_1(i)
+                    ny(nn)=ncfs_all%dim_2(i)
+                    lvlbegin(nn)=k
+                    lvlend(nn)=k
+                    fileid(nn)=myfieldid
+                 enddo
+              else
+                 n3d=n3d+1
+                 mynlvl3d=min(nlvl3d_list(n3d),nz)
+                 nlvlcore=nz/mynlvl3d
+                 j=nz-nlvlcore*mynlvl3d
+                 do k=1,mynlvl3d
+                    nn=nn+1
+                    fileid(nn)=myfieldid
+                    varname(nn)=trim(ncfs_all%list_varname(i))
+                    vartype(nn)=ncfs_all%vartype(i)
+                    nx(nn)=ncfs_all%dim_1(i)
+                    ny(nn)=ncfs_all%dim_2(i)
+                    if(k==1) then
+                       lvlbegin(nn)=1
+                    else
+                       lvlbegin(nn)=lvlend(nn-1)+1
+                    endif
+                    lvlend(nn)=lvlbegin(nn)+nlvlcore-1
+                    if(k<=j) lvlend(nn)=lvlend(nn)+1
+                    if(k==mynlvl3d) lvlend(nn)=min(lvlend(nn),nz)
+                 enddo
+              endif
+           elseif(ncfs_all%num_dim(i)==2) then
               nn=nn+1
-              varname(nn)=trim(ncfs_sfc%list_varname(i))
-              vartype(nn)=ncfs_sfc%vartype(i)
-              nx(nn)=ncfs_sfc%dim_1(i)
-              ny(nn)=ncfs_sfc%dim_2(i)
+              varname(nn)=trim(ncfs_all%list_varname(i))
+              vartype(nn)=ncfs_all%vartype(i)
+              nx(nn)=ncfs_all%dim_1(i)
+              ny(nn)=ncfs_all%dim_2(i)
               lvlbegin(nn)=1
               lvlend(nn)=1
-              fileid(nn)=3
+              fileid(nn)=myfieldid
```

Comment:
Finally, shall we expect the final nn=ntotalcore?  or if it is acceptable if it is a little smaller than ntotalcore (that means some cores are not used)?

Reply:
nn can be smaller than ntotalcore.